src/main/java/org/candlepin/service/exception/subscription/SubscriptionServiceExceptionMapper.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2009 - 2023 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.service.exception.subscription;
+
+import org.xnap.commons.i18n.I18n;
+
+
+
+public class SubscriptionServiceExceptionMapper {
```

Comment:
While I'm not a huge fan of these mappers, if we're going to have them at all, they should all function the same way. This mapper returns a string for the pre-translated message to be, presumably, wrapped in another exception; rather than just throwing the exception like the other mappers do.

So far this seems to be the odd one out, and given the point of abstraction out the handling of specific exceptions is to avoid boilerplate, this class should follow in the convention of returning (not throwing!) the appropriate exception established by the other mappers.

As a side note, since the mappers are throwing Candlepin or API-facing exceptions, these should only ever be used from the API handlers (typically "resource" classes). Additionally, the mappers shouldn't be forced to accept a bunch of state to build the error messages needed for the mapped exceptions -- that bit should be contained within the exception itself. I would expect the mapper call to be nothing more than:

`throw SubscriptionServiceExceptionMapper.map(exception);`

For the exceptions that wrap additional state, that should be present in their constructors. E.g. `throw new SubscriptionExpiredTagException(tag)`